File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        minPrice = float('infinity')
        maxProfit = 0

        for p in prices:
            if p < minPrice:
                minPrice = p
            
            profit = p - minPrice

            if profit > maxProfit:
                maxProfit = profit

        return maxProfit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int minPrice = 999999999;
        int maxProfit = 0;

        for (int i = 0; i < prices.length; i++){
            int p = prices[i];
            if (p < minPrice){
                minPrice = p;
            }
            
            int profit = p - minPrice;

            if (profit > maxProfit){
                maxProfit = profit;
            }
        }

        return maxProfit;
    }
}